cloudify_gcp/admin/projects.py

```diff
@@ -38,46 +48,66 @@ def get(self):
         details retrieval
         """
         self.logger.info('Get instance {0} details'.format(self.name))
-
         return self.discovery.projects().get(
             projectId=self.project_id).execute()
 
     @gcp.check_response
     def create(self):
-        project_body = {
-            constants.NAME: utils.get_gcp_resource_name(
-                ctx.node.properties[constants.NAME]),
-            'projectId': self.project_id
-        }
-        self.logger.info('Project info: {}'.format(repr(project_body)))
-        return self.discovery.projects().create(
-            body=project_body).execute()
+
+        try:
+            resource_exists = self.get()
+            return resource_exists
+        except Exception:
+            project_body = {
+                'name': self.name,
+                'projectId': self.project_id,
+                'parent': self.parent
+            }
+            self.logger.info('Project info: {}'.format(repr(project_body)))
+            result = self.discovery.projects().create(
+                body=project_body).execute()
+            return result
```

Comment:
Instead of doing the exception handling for `get` in `create`, Then `create` can focus just on creating:
```

            return self.discovery.projects().create(
                body=self.project_body).execute()

```